kotlin/src/test/kotlin/feign/kotlin/CoroutineFeignTest.kt

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2012-2022 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign
+
+import com.google.gson.Gson
+import com.google.gson.JsonIOException
+import feign.codec.Decoder
+import feign.codec.Encoder
+import feign.codec.ErrorDecoder
+import kotlinx.coroutines.runBlocking
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import java.io.IOException
+import java.lang.reflect.Type
+import feign.kotlin.CoroutineFeign
+
+class SuspendTest {
+    @Test
+    fun shouldRun1(): Unit = runBlocking {
+        // Arrange
+        val server = MockWebServer()
+        val expected = "Hello Worlda"
+        server.enqueue(MockResponse().setBody(expected))
+        val client = TestInterfaceAsyncBuilder()
+            .target("http://localhost:" + server.port)
+
+        // Act
+        val firstOrder = client.findOrder1(orderId = 1)
+
+        // Assert
+        assertThat(firstOrder).isEqualTo(expected)
+    }
+
+    @Test
+    fun shouldRun2(): Unit = runBlocking {
+        // Arrange
+        val server = MockWebServer()
+        val expected = IceCreamOrder(
+            id = "HELLO WORLD",
+            no = 999,
+        )
+        server.enqueue(MockResponse().setBody("{ id: '${expected.id}', no: '${expected.no}'}"))
+
+        val client = TestInterfaceAsyncBuilder()
+            .decoder(GsonDecoder())
+            .target("http://localhost:" + server.port)
+
+        // Act
+        val firstOrder = client.findOrder2(orderId = 1)
+
+        // Assert
+        assertThat(firstOrder).isEqualTo(expected)
+    }
+
+    @Test
+    fun shouldRun3(): Unit = runBlocking {
+        // Arrange
+        val server = MockWebServer()
+        server.enqueue(MockResponse().setBody("HELLO WORLD"))
+
+        val client = TestInterfaceAsyncBuilder()
+            .target("http://localhost:" + server.port)
+
+        // Act
+        val firstOrder = client.findOrder3(orderId = 1)
+
+        // Assert
+        assertThat(firstOrder).isNull()
+    }
+
+    @Test
+    fun shouldRun4(): Unit = runBlocking {
```

Comment:
could we get meaningful names for tests?

Reply:
Corrected. Please provide feedback on the naming convention.